src/core/throttle.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2024 ScyllaDB
+ */
+
+#ifdef SEASTAR_MODULE
+module;
+#endif
+
+#include <seastar/core/internal/throttle.hh>
```

Comment:
A general question related to the build with modules -- as far as I understand `./src/seastar.cc` includes all headers to provide the declarations and definitions when we build the code with modules.

The source files (e.g. `fair_queue.cc`, `io_queue.cc` and `reactor.cc`) use ifndef e.g.:
```
#ifdef SEASTAR_MODULE
module seastar;
#else
// include the headers related to seastar
#endif
```

The comment from seastar.cc:
```
// include all declaration and definitions to be exported in the the module
// purview
```

My question is as follows -- do we want to follow the same approach in `throttle.cc`? Also, is there an explicit rule in seastar project related to inclusion of files and `SEASTAR_MODULE` macro?

Reply:
I'm not sure if there's any set of rules on how to manipulate SEASTAR_MODULE and related. @tchaikov works on modules support, he can shed more light on this

Reply:
@pwrobelse hi Patryk, thank you for bringing this up.

`seastar.cc` includes both the global module fragment and the purview of the `seastar` module. we put the implementation into the module implementation units which starts with `module seastar`, for instance, see `fair_queue.cc` as you pointed out.

when it comes to `throttle.cc`, it needs to access the seastar module, so i think we would need to start it with `module seastar;` instead of `#include <seastar/core/internal/throttle.hh>` when C++20 modules is enabled, which is checked using `#ifdef SEASTAR_MODULE` in general.

> I'm not sure if there's any set of rules on how to manipulate SEASTAR_MODULE and related.

@xemul good point. i will note down this. could you suggest which document i should put this in? as its target audience will be library developer, instead of library user, the closest one i can find is `coding-style.md`. but this document focuses on coding style, it's more about taste, instead of the guideline on the general design or a perspective of the code structure. shall i create another document on C++20 modules?